src/opera/pge/dist_s1/dist_s1_pge.py

```diff
@@ -37,6 +39,143 @@ class DistS1PreProcessorMixin(PreProcessorMixin):
+        4. Ensures input RTCs are not from a mixture of Sentinel-1A and Sentinel-1C
+        5. Validates that the co- and cross-pol RTCs are in the same order (burst-ID &
+           acquisition time)
+        6. Validates no cross-pol RTCs are in copol input lists and vice-versa
+        """
+
+        sas_config = self.runconfig.sas_config
+
+        pre_rtc_copol = sas_config["run_config"]["pre_rtc_copol"]
+        pre_rtc_crosspol = sas_config["run_config"]["pre_rtc_crosspol"]
+        post_rtc_copol = sas_config["run_config"]["post_rtc_copol"]
+        post_rtc_crosspol = sas_config["run_config"]["post_rtc_crosspol"]
+
+        if len(pre_rtc_copol) != len(pre_rtc_crosspol) or len(post_rtc_copol) != len(post_rtc_crosspol):
+            msg = (f'Lengths of input pre/post co/cross pol input RTC lists differ: pre {len(pre_rtc_copol)} '
+                   f'{len(pre_rtc_crosspol)} post {len(post_rtc_copol)} {len(post_rtc_crosspol)}')
+            self.logger.critical(
+                self.name,
+                ErrorCode.INVALID_INPUT,
+                msg
+            )
+
+        all_rtcs = pre_rtc_copol + pre_rtc_crosspol + post_rtc_copol + post_rtc_crosspol
+
+        if not set([basename(rtc) for rtc in all_rtcs]).issubset([basename(rtc) for rtc in self.runconfig.input_files]):
+            msg = 'SAS RTC file groups do not make a subset of PGE Input Files'
+            self.logger.critical(
+                self.name,
+                ErrorCode.INVALID_INPUT,
+                msg
+            )
+
+        rtc_pattern = re.compile(r'(?P<id>(?P<project>OPERA)_(?P<level>L2)_(?P<product_type>RTC)-(?P<source>S1)_'
+                                 r'(?P<burst_id>\w{4}-\w{6}-\w{3})_(?P<acquisition_ts>(?P<acq_year>\d{4})(?P<acq_month>'
+                                 r'\d{2})(?P<acq_day>\d{2})T(?P<acq_hour>\d{2})(?P<acq_minute>\d{2})(?P<acq_second>'
+                                 r'\d{2})Z)_(?P<creation_ts>(?P<cre_year>\d{4})(?P<cre_month>\d{2})(?P<cre_day>\d{2})T'
+                                 r'(?P<cre_hour>\d{2})(?P<cre_minute>\d{2})(?P<cre_second>\d{2})Z)_'
+                                 r'(?P<sensor>S1A|S1B|S1C)_(?P<spacing>30)_(?P<product_version>v\d+[.]\d+))'
+                                 r'(_(?P<pol>VV|VH|HH|HV|VV\+VH|HH\+HV)|_BROWSE|_mask)?[.]'
+                                 r'(?P<ext>tif|tiff|h5|png|iso\.xml)$')
+
+        rtc_matches = [rtc_pattern.match(basename(rtc)) for rtc in all_rtcs]
+
+        if any([match is None for match in rtc_matches]):
+            msg = 'Invalid RTC filenames in SAS input'
+            self.logger.critical(
+                self.name,
+                ErrorCode.INVALID_INPUT,
+                msg
+            )
+
+        if len(set([match.groupdict()['sensor'] for match in rtc_matches])) > 1:
+            msg = 'SAS input contains a RTCs from more than one S1 Sensor. Inputs should be all from S1A or S1C'
```

Comment:
Let's reword this first sentence to `RunConfig contains RTCs from more than one S1 Sensor.`